TACC/Core-CMS-Custom
# task/WP-204:  Add type of plan checklist to registration entities

## Overview
Added a new checklist to the registration form to account for new fields on registration entities, and added needed fields & HTML everywhere else relevant.
…

## Related

- [WP-204](https://jira.tacc.utexas.edu/browse/WP-204)
…

## Changes
- Added checklist to registration form (for new registration form and edit registration modal)
- Added list to show chosen plan types on View Registration modal
- Added relevant database fields to all entity database utilities
- Updated some styling for this checklist to display correctly on form
…

## Testing

1. Go to http://localhost:8000/register/request-to-submit/
2. Fill out form as normal, and try to submit form _without_ checking any of the plan type options
3. Ensure there is a message asking you to select at least one of the options
4. Confirm form submits successfully
5. Go to http://localhost:8000/administration/list-registration-requests
6. Find the record you just submitted, and confirm that:
    - In the View Record modal, the list of options only you selected for plan type appears
    - In the Edit Record modal, the plan type checklist correctly shows your previous choices as selected
    - From above, deselect all choices you made, and confirm that you cannot submit the edit action with no plan types selected

## UI

…
